scripts/globals/goblinfootprint.lua

```diff
@@ -0,0 +1,218 @@
+        [16] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [17] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [18] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [19] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [20] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [21] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [22] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [23] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [24] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [25] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+        [26] = {},              -- QUEST NOT IMPLEMENTED (ROV)
+    },
+    [dsp.zone.NORTH_GUSTABERG] =
+    {
+        [1] = {200, 1092},      -- As Thick as Thieves
+        [2] = {22},             -- The Gustaberg Tour
+        [3] = {254},            -- Apocalypse
+        [4] = {0},              -- An Ulcerous Uragnite (pt.1)
+        [5] = {1},              -- An Ulcerous Uragnite (pt.2)
+    },
+    [dsp.zone.SOUTH_GUSTABERG] =
+    {
+        [1] = {906},            -- A Transient Dream
+        [2] = {0},              -- A Beaked Blusterer (pt.1)
+        [3] = {1},              -- A Beaked Blusterer (pt.2)
+    },
+    [dsp.zone.MERIPHATAUD_MOUNTAINS] =
+    {
+        [1] = {33},              -- The Holy Crest
+    },
+    [dsp.zone.ROMAEVE] =
+    {
+        [1] = {2},              -- Vain
+        [2] = {4},              -- Moon Reading
+        [3] = {7, 917, 1408},   -- Divine Might
+        [4] = {71},             -- Fountain of Trouble
+    },
+    [dsp.zone.CASTLE_ZVAHL_KEEP] =
+    {
+        [1] = {8},
+    },
+    [dsp.zone.BOSTAUNIEUX_OUBLIETTE] =
+    {
+        [1] = {13},             -- The Rumor (pt. 1)
+        [2] = {12},             -- The Rumor (pt. 2)
+        [3] = {17},             -- Trouble at the Sluice
+        [4] = {6},              -- The Holy Crest
+        [5] = {0},              -- Souls in Shadows (pt. 1)
+        [6] = {1},              -- Souls in Shadows (pt. 2)
+        [7] = {5},              -- Souls in Shadows (pt. 3)
+    },
+    [dsp.zone.MAZE_OF_SHAKHRAMI] =
+    {
+        [1] = {66}              -- Equipped for All Occasions
+    },
+    [dsp.zone.GARLAIGE_CITADEL] =
+    {
+        [1] = {60},             -- Escort for Hire (Windurst)
+        [2] = {14},             -- Peace for the Spirit
+    },
+}
+
+function dsp.goblinfootprint.rewatch(player, trigger)
+    local zone = player:getZoneID()
+    local ID = zones[zone]
+    local options = 1
+
+    for i, check in pairs(csReq[zone]) do
+        if check(player) == true then
+            options = bit.bor(options, bit.lshift(1, i))
+            if trigger then
+                player:messageSpecial(ID.text.ITEMS_ITEMS_LA_LA)
+                return
+            end
+        end
+    end
+    if options > 1 then
+        player:startEvent(gobCS[zone], bit.bnot(options), -2, -2, -2)
```

Comment:
What are these -2s? params are unsigned values. Is this from a retail capture or just unimplemented params? If it's unimplemented params, just leave a comment. If it's from the capture, it could just be garbage data. Retail does not zero out packets before sending them (dsp does). I would recommend trying to remove these extra params and see if the Goblins still work.

Reply:
These -2's are the 4294967294 value he had there before that are [apparently required](https://github.com/DarkstarProject/darkstar/pull/6230#discussion_r330840136). I just figured -2 looked cleaner.

Reply:
these are the 0xFFFFFFFE he explained previously, which is preferred if it must be that precise value.

https://github.com/DarkstarProject/darkstar/pull/6230#discussion_r330846095

and not just certain bits that value happened to cover

Reply:
@nfaletra those extra params are needed for the choices to populate on the !cs just like @cocosolos explained. He helped me make the function look cleaner as when I originally when I started this PR i had a horrible bit table as an array (you can check how horrible it was if you see first commit lol)

Reply:
@TeoTwawki thanks for the reminder :)
Yeah those -2 are a static value, it's not used or change in the script to present less or more choices

Reply:
A quick comment above that line is fine, then. Thanks for breaking it down. I may investigate on retail further at some point.